flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskMailboxTestHarness.java

```diff
@@ -122,6 +122,12 @@ private void maybeProcess() throws Exception {
 
 	public void process() throws Exception {
 		while (streamTask.inputProcessor.isAvailable() && streamTask.mailboxProcessor.isMailboxLoopRunning()) {
+			processSingleStep();
```

Comment:
Kind of replicates the checks. Would it make sense to have a `boolean processSingleStep`, where we can just execute a `while(processSingleStep()) ;`?